Plots/Polygons/NCL_polyg_4.py

```diff
@@ -0,0 +1,230 @@
+                       ]
+    ax.clabel(hdl,
+              np.arange(-8, 24, 8),    # Only label these contour levels: [-8, 0, 8, 16]
+              fontsize="small",
+              colors="black",
+              fmt="%.0f",              # Turn off decimal points
+              manual=label_locations,  # Manual label locations
+              inline=False)            # Don't remove the contour line where labels are located.
+
+    # Create a rectangle patch, to color the border of the rectangle a different color.
+    # Specify the rectangle as a corner point with width and height, to help place border text more easily.
+    left, width = -90, 45
+    bottom, height = 0, 30
+    right = left + width
+    top = bottom + height
+
+    p = plt.Rectangle((left, bottom), width, height, fill=False,
+                      zorder=3,          # Plot on top of the purple box border.
+                      edgecolor='red',
+                      alpha=0.5)         # Lower color intensity.
+    ax.add_patch(p)
+
+    # Draw text labels around the box.
+    # Change the default padding around a text box to zero, making it a "tight" box.
+    # Create "text_args" to keep from repeating code when drawing text.
+    text_shared_args = dict(
+        fontsize=8,
+        bbox=dict(boxstyle='square, pad=0', facecolor='white', edgecolor='white'),
+    )
+
+    # Draw top text
+    ax.text(left + 0.6 * width, top, 'test',
+            horizontalalignment='right',
+            verticalalignment='center',
+            **text_shared_args
+            )
+
+    # Draw bottom text.   Change text background to match the map.
+    ax.text(left + 0.5 * width, bottom, 'test',
+            horizontalalignment='right',
+            verticalalignment='center',
+            fontsize=8,
+            bbox=dict(boxstyle='square, pad=0', facecolor='lightgrey', edgecolor='lightgrey'),
+            )
+
+    # Draw left text
+    ax.text(left, top, 'test',
+            horizontalalignment='center',
+            verticalalignment='top',
+            rotation=90,
+            **text_shared_args
+            )
+
+    # Draw right text
+    ax.text(right, bottom, 'test',
+            horizontalalignment='center',
+            verticalalignment='bottom',
+            rotation=-90,
+            **text_shared_args
+            )
+
+    # Add title and tick marks to match NCL conventions.
+    ax.text(0.0,  1.05, "Zonal Wind", fontsize=12, transform=ax.transAxes)
+    ax.text(1.0,  1.05, "m/s", fontsize=12, transform=ax.transAxes, horizontalalignment='right')
+    ax.text(1.0, -0.20, "CONTOUR FROM -12 TO 40 BY 4",
+            fontname='Helvetica',
+            horizontalalignment='right',
+            transform=ax.transAxes,
+            bbox=dict(boxstyle='square, pad=0.1', facecolor='white', edgecolor='black'))
+
+    gcv.util.nclize_axis(ax)
+    gcv.util.add_lat_lon_ticklabels(ax)
+
```

Comment:
After these two lines, I would add

```python
ax.yaxis.set_minor_locator(AutoMinorLocator(n=4))
```

To get the right number of minor tick marks in the y axis (especially nice because your major ticks are 20 degrees apart, so this makes the minor ticks 5 degrees apart instead of 6 2/3 degrees). You'll also need

```
from matplotlib.ticker import AutoMinorLocator
```

in your `import` block

Reply:
@mnlevy1981  That's a good catch, thanks Mike!   I've pushed the change.